pkg/controller/cluster_controller.go

```diff
@@ -99,6 +123,13 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		// Ensure the action is applicable to the current resource state
 		if a.Handles(cluster.Status().Conditions) {
 			if err := a.Act(ctx, &cluster); err != nil {
+				log.Info("Error on action", "Action", a.GetActionType(),"err", err.Error())
+				cluster.SetActionFailed(a.GetActionType(), err.Error())
+				defer func(ctx context.Context, cluster *resource.Cluster) {
```

Comment:
We need documentation on why we are doing this

Reply:
I added a comment  for this